Cleanup: refactor deeply nested if statements

Rakshith-R opened the pull request:
# Describe what this PR does #

Refactors  deeply nested if statements to reduce cognitive complexity.

## Related issues ##

Closes: #1229 

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

- `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)
- `/retest all`: run this in case the CentOS CI failed to start/report any test
  progress or results

</details>

nixpanic commented on cmd/cephcsi.go:
```diff
@@ -148,14 +148,13 @@ func main() {
-				util.DefaultLog("Reconfigured PID limit to %d%s", conf.PidLimit, s)
+			case conf.PidLimit == -1:
+				util.DefaultLog("Reconfigured PID limit to %d (max)", conf.PidLimit)
+			case conf.PidLimit != -1:
```
instead of this case, you can use `default:`

nixpanic commented on internal/util/vault_tokens.go:
```diff
@@ -506,3 +489,27 @@ func (kms *VaultTokensKMS) parseTenantConfig() error {
+	if tenant == "" {
+		return nil
+	}
+	kms.Tenant = tenant
```
a function to fetch the tenant-config, should not set anything in the `kms` object.

nixpanic commented on internal/util/vault_tokens.go:
```diff
@@ -506,3 +489,27 @@ func (kms *VaultTokensKMS) parseTenantConfig() error {
 }
+
+// fetchTenantConfig fetches the configuration for the tenant if it exists.
+func fetchTenantConfig(kms *VaultTokensKMS, config map[string]interface{}, tenant string) error {
```
instead of passing `kms`, make it a method of the VaultTokensKMS type

nixpanic commented on internal/journal/voljournal.go:
```diff
@@ -310,10 +310,12 @@ func (conn *Connection) CheckReservation(ctx context.Context,
 
 	// check UUID only encoded value
-	if len(objUUIDAndPool) == uuidEncodedLength {
+	switch len(objUUIDAndPool) {
```
I do not think this reduces the complexity, it may make the code even more complex. Here a simple if/else is sufficient and simple enough. This applies to other locations too.

nixpanic commented on internal/journal/voljournal.go:
```diff
@@ -409,11 +411,11 @@ func (conn *Connection) UndoReservation(ctx context.Context,
+	case volName != "" && len(volName) < uuidEncodedLength:
+		return fmt.Errorf("unable to parse UUID from %s, too short", volName)
 
+	case volName != "":
```
I think the if-statement was fine here. Can you explain why the switch/select is better?

nixpanic commented on e2e/utils.go:
```diff
@@ -401,17 +401,18 @@ func pvcDeleteWhenPoolNotFound(pvcPath string, cephfs bool, f *framework.Framewo
 		return err
 	}
-	if cephfs {
+	switch cephfs {
```
this is jusn an if/else, that should not be more complex than a switch/case (that does not have a `default:`)?

Rakshith-R committed:
cleanup: refactor deeply nested if statement in cephcsi.go

Refactored deeply nested if statement in cephcsi.go to
reduce cognitive complexity.

Signed-off-by: Rakshith R <rar@redhat.com>

Rakshith-R committed:
cleanup: refactor deeply nested if statements in persistentvolume.go

Refactored deeply nested if statement in persistentvolume.go to
reduce cognitive complexity.

Signed-off-by: Rakshith R <rar@redhat.com>

Rakshith-R committed:
cleanup: refactor deeply nested if statements in internal/cephfs

Refactored deeply nested if statement in internal/cephfs to
reduce cognitive complexity.

Signed-off-by: Rakshith R <rar@redhat.com>

Rakshith-R committed:
cleanup: refactor deeply nested if statement in vault_tokens.go

Refactored deeply nested if statement in vault_tokens.go to
reduce cognitive complexity by adding fetchTenantConfig function.

Signed-off-by: Rakshith R <rar@redhat.com>

Rakshith-R committed:
cleanup: refactor deeply nested if statements in internal/rbd

Refactored deeply nested if statement in internal/rbd to
reduce cognitive complexity.

Signed-off-by: Rakshith R <rar@redhat.com>

Rakshith-R committed:
ci: enable nestif linter

The nestif linter reports deeply nested if statements.
This commit enables the nestif linter and sets
min-complexity to 7.

Closes: #1229

Signed-off-by: Rakshith R <rar@redhat.com>

Rakshith-R commented:
/retest ci/centos/upgrade-tests-cephfs 
/retest ci/centos/upgrade-tests-rbd

Rakshith-R commented:
/retest ci/centos/upgrade-tests-cephfs

Rakshith-R commented:
/retest ci/centos/upgrade-tests-rbd

Rakshith-R commented:
> /retest ci/centos/upgrade-tests-rbd
> /retest ci/centos/upgrade-tests-cephfs

Failed due to docker pull rate limiter

humblec commented on internal/rbd/nodeserver.go:
```diff
@@ -531,30 +530,30 @@ func (ns *NodeServer) mountVolume(ctx context.Context, stagingPath string, req *
+			return notMnt, status.Error(codes.Internal, err.Error())
+		}
+	} else {
+		// Create a directory
```
may be mention, create a mountpath or create a mountpath directory ?

humblec commented on internal/controller/persistentvolume/persistentvolume.go:
```diff
@@ -129,43 +129,44 @@ func (r ReconcilePersistentVolume) reconcilePV(obj runtime.Object) error {
-		defer cr.DeleteCredentials()
+	cr, err := r.getCredentials(secretName, secretNamespace)
+	if err != nil {
+		util.ErrorLogMsg("failed to get credentials %s", err)
```
failed to get credentials from secret would be more appropriate.

nixpanic reviewed (approved):
Great, thanks!

Rakshith-R commented:
/retest ci/centos/mini-e2e/k8s-1.19

Rakshith-R commented:
/restest ci/centos/upgrade-tests-cephfs

Rakshith-R commented:
/retest ci/centos/upgrade-tests-cephfs

Rakshith-R commented:
/retest ci/centos/upgrade-tests-cephfs

Rakshith-R commented:
> /retest ci/centos/mini-e2e/k8s-1.19
> /retest ci/centos/upgrade-tests-cephfs


`Error from server: error when creating "https://raw.githubusercontent.com/rook/rook/v1.4.9/cluster/examples/kubernetes/ceph/toolbox.yaml": etcdserver: request timed out`

Rakshith-R commented:
/retest ci/centos/upgrade-tests-cephfs

nixpanic commented:
/retest ci/centos/mini-e2e/k8s-1.19

nixpanic commented:
> /retest ci/centos/mini-e2e/k8s-1.19

Deploying Rook failed #1636

nixpanic commented:
/retest ci/centos/upgrade-tests-cephfs

nixpanic commented:
> /retest ci/centos/upgrade-tests-cephfs

minikube VM timed out:
```
Waiting up to 3m0s for all (but 0) nodes to be ready
```

Rakshith-R commented:
/retest ci/centos/mini-e2e/k8s-1.19

Rakshith-R commented:
/retest ci/centos/mini-e2e-helm/k8s-1.20
